src/CodeCollection/CodeChangeSet.php

```diff
@@ -115,45 +188,111 @@ public function hasWarnings($path)
      * Return the file contents for a given path
      *
      * @param string $path
-     * @return string
-     * @throws \InvalidArgumentException
+     * @return string|false
+     * @throws InvalidArgumentException If `$path` has not been recorded as changed.
      */
-    public function newContents($path)
+    public function newContents(string $path)
     {
-        if (isset($this->fileChanges[$path])) {
-            return $this->fileChanges[$path]['new'];
-        } else {
-            throw new \InvalidArgumentException("No file changes found for $path");
-        }
+        $change = $this->changeByPath($path);
+        return isset($change['new']) ? $change['new'] : false;
     }
 
     /**
      * Return the prior file contents for a given path
      *
      * @param string $path
+     * @return string|false
+     * @throws InvalidArgumentException If `$path` has not been recorded as changed.
+     */
+    public function oldContents(string $path)
+    {
+        $change = $this->changeByPath($path);
+        return isset($change['old']) ? $change['old'] : false;
+    }
+
+    /**
+     * Return the new Path for a file.
+     * @param string $path
+     * @return string|false
+     * @throws InvalidArgumentException If `$path` has not been recorded as changed.
+     */
+    public function newPath(string $path)
+    {
+        return $this->changeByPath($path)['path'];
+    }
+
+    /**
+     * Return the operation that should be applied for the provided path. Will be one of:
+     * * `modified` for file with updated content,
+     * * `renamed` for files or folder that should be moved to a different location, with or without modifications,
+     * * `deleted` for files or folder that should be deleted,
+     * * `` for files that don't appear to have any outstanding operation against them.
+     * @param string $path
      * @return string
-     * @throws \InvalidArgumentException
      */
-    public function oldContents($path)
+    public function opsByPath(string $path): string
     {
         if (isset($this->fileChanges[$path])) {
-            return $this->fileChanges[$path]['old'];
-        } else {
-            throw new \InvalidArgumentException("No file changes found for $path");
+            $change = $this->fileChanges[$path];
+
+            if ($change['path'] === false) {
+                // If the path attribute is false, we are deleting the file
+                return 'deleted';
+            } elseif ($change['path'] != $path) {
+                // If the change path is different than the key path we are moving the file.
+                return 'renamed';
+            }
+
+            if ($this->hasNewContents($path)) {
+                // If we have new contents with old content, we are modifying a file. Otherwise, it's a new file.
+                return (isset($change['old']) && $change['old'] !== false) ?
+                    'modified':
+                    'new file';
+            }
         }
+
```

Comment:
I'd prefer to see these ops assigned to constants. Just to make it more robust and expressive in your IDE.

Reply:
Done.